src/components/button.js

```diff
@@ -97,8 +97,13 @@ const Button = ({ children, classes, variant, size, type, disabled, ...rest }) =
     endIcon: `${sizes.iconPadding.endIcon[size]} ${classes.endIcon}`,
   }
 
+  const handleClick = e => {
+    e.stopPropagation()
+    if('onClick' in rest) rest.onClick()
```

Comment:
we need pass down event to onClick() otherwise user won't have access to it

Reply:
Would this be more appropriate solution:

`  const handleClick = e => {
    e.stopPropagation()
    if ('onClick' in rest && typeof rest.onClick === 'function') {
      console.log(e)
      rest.onClick(e)
    }
  }
`

Reply:
you can simplify the condition by a bit to one

```js
typeof rest?.onClick === 'function'
```